### PR TITLE
DS eval for OPT on WikiText

### PR DESCRIPTION
**Pull request by natuan**
This PR enables verifying perplexity of OPT models on wikitext, matching the numbers in pytorch eval flow after one-shot prune/quantize with SparseGPT.

Qualification: tested with 1.3, 2.7b base, W8A8linear and pruned50_quantW8A8 models.

### CODE REVIEW

File: src/deepsparse/transformers/eval_downstream.py [hunk 1]
```diff
@@ -109,6 +112,82 @@ def perplexity_eval(args, batch_size=16, dataset_name="openai_humaneval"):
     return perplexity_metrics
 
 
+def perplexity_wikitext_eval(args, batch_size=16, dataset_name=""):
```
Comment (dbogunowicz) [inline src/deepsparse/transformers/eval_downstream.py:115]: why are we creating a duplicate of `perplexity_eval` function? i do not see a reason why we should have a separate function, that mostly is just adding boilerplate code.

File: src/deepsparse/transformers/metrics.py [hunk 2]
```diff
@@ -122,22 +122,6 @@ def add_batch(self, predictions: List[str]):
 
             logits = out.logits
 
-            if not self._pipeline.has_cache:
```
Comment (dbogunowicz) [inline src/deepsparse/transformers/metrics.py:125]: why are we moving this logic from the metric to the evaluation script? this does not make sense to me from the design point of view and would only add unnecessary code.

Comment (dbogunowicz) [inline src/deepsparse/transformers/metrics.py:125]: `has_cache` has been renamed to `cache_support_enabled` btw

**Review by dbogunowicz — CHANGES_REQUESTED** [review]
Let's clean up the code, there is no reason to write so much unnecessary lines.

File: src/deepsparse/transformers/metrics.py (hunk 2 above)
Comment (natuan) [reply src/deepsparse/transformers/metrics.py:125]: I tried to follow the ppl and metrics design here, but got bad results. I did it this way to match how it was done from the pytorch flow, and mainly to make sure we got the right ppl numbers; plus for the sake of time. If you have concrete suggestions how to move these into the framework, please help.

File: src/deepsparse/transformers/eval_downstream.py (hunk 1 above)
Comment (natuan) [reply src/deepsparse/transformers/eval_downstream.py:115]: Tried to reuse this func, swapping in wikitext, got bad results.

File: src/deepsparse/transformers/metrics.py (hunk 2 above)
Comment (abhinavnmagic) [reply src/deepsparse/transformers/metrics.py:125]: I think this pathway is important to support perplexity evaluation of non-cache models. Wouldn't we include likelihood for pad tokens if we remove this? For testing, did you test non-cache models?

**Comment by natuan** [comment]
Closing because this PR will be covered [here](https://github.com/neuralmagic/deepsparse/pull/1197)
